codecov.yml

```diff
@@ -7,3 +7,9 @@ coverage:
     patch:
       default:
         target: 80%
+        paths:
+          - "!controllers/provider"
+      credentials:
+        paths:
+          - controllers/provider
+        target: 40%
```

Comment:
Let's remove these lines. For the first time of the contribution, it's okay not to meet the target:)

Reply:
> This implementation is interesting. Have you built your own Terraform docker image https://github.com/oam-dev/terraform-controller/blob/master/chart/values.yaml#L10?

Not yet. I'm using private registries

Reply:
But you still need a self-built Terraform docker image?

Reply:
I hope to have a whole story on how to use the custom provider, and others can use the feature you contributed.

Reply:
I still use the terrain from image provided by OAM.

When terraform is applied, the provider defined in HCl will be pulled.

The custom provider (credentials) is used to solve the problem of private provider authentication. The existing credentials field cannot meet my needs. At the same time, I do not want to disclose my authentication field and other extended fields (business types, e.g: COST_ACCOUNT)

Reply:
Thanks, I got the point. 

One last question, can you post an example of your Configuration (especially your HCL/.tf) without any sensitive information?

Reply:
Ok, let me add that.